src/pages/index.js

```diff
@@ -1,38 +1,40 @@
 // ***********************************   DECLARING VARIABLES   ************************************************************
 //IMPORTS
 import './index.css';
-import {initialCards, editButton, profileNameClass, profileDescriptionClass, addPlaceButton, cardModalForm, profileModalForm, cardContainerClass} from '../utils/constants.js';
-import Section from '../components/Section.js';
-import {makePopUpVisible, makeAddPlacePopUpVisible, createNewUserInfo, createFormValidator, createPopupWithForm, createPopupWithImage, createNewCard} from '../utils/utils.js';
+import {editButton, profileNameClass, profileDescriptionClass, addPlaceButton, cardModalForm, profileModalForm, avatarModalForm, avatarButton, profileModal, cardModal, avatarModal} from '../utils/constants.js';
+import {makePopUpVisible, makeAddPlacePopUpVisible, createNewUserInfo, createFormValidator, createPopupWithForm, createPopupWithImage, makeAvatarPopUpVisible} from '../utils/utils.js';
+import Api from '../components/Api.js'
 
 
+const api = new Api();
+api.loadUserInfo();
+api.loadCards();
+
 
 //STORING CLASS INSTANCES
 const imagePopup = createPopupWithImage('#card-modal');
 const userInfo = createNewUserInfo(profileNameClass, profileDescriptionClass);
 const formValidatorProfileModal = createFormValidator(profileModalForm);
 const formValidatorCardModal = createFormValidator(cardModalForm);
-const addPlacePopup = createPopupWithForm('#add-card-modal', (inputValues) => {
-  const titleValue = inputValues.title;
-  const urlValue = inputValues.url;
-  
-  const cardElement = createNewCard({name: titleValue, link: urlValue});
+const formValidatorAvatarModal = createFormValidator(avatarModalForm);
+
 
-  renderCards.prependItem(cardElement);
-  formValidatorCardModal.disableStateButton();
+const addPlacePopup = createPopupWithForm('#add-card-modal', (inputValues) => {
+  api.renderLoading(true, cardModal);
+  api.addNewCard(inputValues, cardModal);
 });
 
+
 const userInfoPopup = createPopupWithForm('#profile-modal', (inputValues) => {
-  userInfo.setUserInfo(inputValues);
+  api.renderLoading(true, profileModal);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```